internals/api/uuid/uuid.go

```diff
@@ -39,3 +47,18 @@ func (u *UUID) IsZero() bool {
 func Equal(a *UUID, b *UUID) bool {
 	return gid.Equal(a.UUID, b.UUID)
 }
+
+// Validate validates a uuid string is a valid UUID.
+func Validate(str string) error {
+	_, err := FromString(str)
+	return err
+}
+
+// IsErrInvalidUUID returns whether the given error is returned for an invalid uuid.
+func IsErrInvalidUUID(err error) bool {
```

Comment:
Wouldn't it make more sense to have a more general `apiErrorEqual(err1, err2 error) bool` that checks whether the namespace and code of an API error are equal? Then we can reuse that in other cases.